codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/DafnyToSmithyShapeVisitor.java

```diff
@@ -517,47 +520,69 @@ public String integerShape(IntegerShape shape) {
   @Override
   public String longShape(LongShape shape) {
     writer.addImportFromModule("github.com/dafny-lang/DafnyRuntimeGo", "dafny");
-    return (
-      """
-      func() *int64 {
-          var b int64
-          if %s == nil {
-              return nil
-          }
-          b = %s.(int64)
-          return &b
-      }()"""
-    ).formatted(dataSource, dataSource);
+    if ((boolean) isOptional) {
```

Comment:
Isn't `isOptional` already boolean?

Reply:
Yes, this was an existing code. I updated it.